src/metabase/query_processor/middleware/results_metadata.clj

```diff
@@ -35,6 +35,15 @@
     (catch Throwable e
       (log/error e (tru "Error recording results metadata for query")))))
 
+(defn- remove-remapped-columns
+  "Fields with remapped_from attribute are not actual fields that exists in the table,
+  they were added to :cols by [[metabase.query-processor.middleware.add-dimension-projections/remap-results]] middleware.
+
+  We don't want to preserve these fields in the `results_metadata` because some pre-processing middleware like `source-metadata->fields`
+  or `resolve-joins` assume all fields in `results_metadata` are fields exists in DB (see #23449)"
+  [metadata]
+  (filter #(empty? (:remapped_from %)) metadata))
```

Comment:
could this be more simply `(remove :remapped_from metadata)`? then it's almost exactly what the name of the function is.

Reply:
yep!